sdk/js/index.ts

```diff
@@ -20,78 +20,182 @@ export interface StreamInfo {
         functionName: "time_left",
         args: [this.creator, this.streamId],
-      })) as bigint,
+      })) as bigint
     );
   }
 
   async streamInfo(): Promise<StreamInfo> {
     return (await this.publicClient.readContract({
-      address: this.address,
+      address: this.streamManager.address,
       abi: StreamManagerContractType.abi as Abi,
       functionName: "streams",
       args: [this.creator, this.streamId],
     })) as StreamInfo;
   }
 
-  async token(): Promise<Address> {
-    return (await this.streamInfo()).token;
-  }
-
-  async totalTime(): Promise<number> {
+  async totalTime(): Promise<bigint> {
     const streamInfo = await this.streamInfo();
     return (
-      Number(streamInfo.funded_amount / streamInfo.amount_per_second) +
-      Number(streamInfo.last_pull - streamInfo.start_time)
+      streamInfo.funded_amount / BigInt(streamInfo.amount_per_second) +
+      (streamInfo.last_pull - streamInfo.start_time)
     );
   }
+
+  async addTime(amount: bigint): Promise<string> {
+    if (!this.walletClient || !this.walletClient.account)
+      throw new Error("Error funding stream: no wallet client set");
+
+    return this.walletClient.writeContract({
+      chain: null,
+      address: this.streamManager.address,
+      abi: StreamManagerContractType.abi as Abi,
+      functionName: "add_funds",
+      args: [this.creator, this.streamId, amount],
+      account: this.walletClient.account.address,
+    });
+  }
+
+  async cancel(reason?: string): Promise<string> {
+    if (!this.walletClient || !this.walletClient.account)
+      throw new Error("Error cancelling stream: wallet client is not set");
+
+    if (
+      this.walletClient.account.address != this.creator &&
+      this.walletClient.account.address != (await this.streamManager.owner())
+    )
+      // Both the owner and the creator of the stream can cancel it
+      throw new Error(
+        "Error cancelling stream: you are neither the creator nor the owner of the stream."
+      );
+
+    // pass args depending on each situation; reason is optional but must be provided
+    // if not using the caller as the creator arg
+    const args =
+      this.walletClient.account.address != this.creator
+        ? [this.streamId, reason || "", this.creator]
+        : reason
+        ? [this.streamId, reason]
+        : [this.streamId];
+
+    return this.walletClient.writeContract({
+      chain: null,
+      address: this.streamManager.address,
+      abi: StreamManagerContractType.abi as Abi,
+      functionName: "cancel_stream",
+      args: args,
+      account: this.walletClient.account.address,
+    });
+  }
+
+  async isCancelable(): Promise<boolean> {
+    return (await this.publicClient.readContract({
+      address: this.streamManager.address,
+      abi: StreamManagerContractType.abi as Abi,
+      functionName: "stream_is_cancelable",
+      args: [this.creator, this.streamId],
+    })) as boolean;
+  }
```

Comment:
can actually make this sync if `StreamInfo.start_time` is added to the properties on the constructor of `Stream`
```suggestion
  isCancelable(): boolean {
    return this.streamManager.MIN_STREAM_LIFE < (Date.now() / 1000 - this.startTime);
  }
```

Reply:
moved to #62